lib/services/hidden_service.dart

```diff
@@ -23,24 +25,70 @@ extension HiddenService on CollectionsService {
   // getDefaultHiddenCollection will return null if there's no default
   // collection
   Future<Collection> getDefaultHiddenCollection() async {
+    Collection? defaultHidden;
     if (cachedDefaultHiddenCollection != null) {
       return cachedDefaultHiddenCollection!;
     }
     final int userID = config.getUserID()!;
-    final Collection? defaultHidden =
-        collectionIDToCollections.values.firstWhereOrNull(
-      (element) => element.isDefaultHidden() && element.owner!.id == userID,
-    );
+    final allDefaultHidden = collectionIDToCollections.values
+        .where(
+          (element) => element.isDefaultHidden() && element.owner!.id == userID,
+        )
+        .toList();
+
+    if (allDefaultHidden.length > 1) {
+      defaultHidden = await clubAllDefaultHiddenToOne(
+        allDefaultHidden,
+      );
+    } else if (allDefaultHidden.length == 1) {
+      defaultHidden = allDefaultHidden.first;
+    }
+
     if (defaultHidden != null) {
       cachedDefaultHiddenCollection = defaultHidden;
       return cachedDefaultHiddenCollection!;
     }
+
     final Collection createdHiddenCollection =
         await _createDefaultHiddenAlbum();
     cachedDefaultHiddenCollection = createdHiddenCollection;
     return cachedDefaultHiddenCollection!;
   }
 
+  Future<Collection> clubAllDefaultHiddenToOne(
+    List<Collection> allDefaultHidden,
+  ) async {
+    final Collection result = allDefaultHidden.first;
+
+    for (Collection defaultHidden in allDefaultHidden) {
+      try {
+        if (defaultHidden == allDefaultHidden.first) {
+          continue;
+        }
+        final filesInCollection = (await FilesDB.instance.getFilesInCollection(
+          defaultHidden.id,
+          galleryLoadStartTime,
+          galleryLoadEndTime,
+        ))
+            .files;
+        await move(result.id, defaultHidden.id, filesInCollection);
+        CollectionsService.instance.trashEmptyCollection(defaultHidden);
+      } on AssertionError catch (e) {
+        _logger.severe("Clubbing all defaultHidden : ${e.message}");
```

Comment:
why is there special handling for AssertionError?

Reply:
Thought logging would be more clear in case something goes wrong. Realised that it doesn't make a difference. Have removed it.